.github/workflows/code-coverage-linux.yml

```diff
@@ -0,0 +1,85 @@
+name: CI Pipeline Code Coverage Linux
+
+on:
+  pull_request:
+    branches: [ main ]
+
+  # Allow manually triggering from the actions tab
+  workflow_dispatch:
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Debug
+
+  # Enable colour output for GTEST and XML file generation
+  GTEST_COLOR: yes
+  GTEST_OUTPUT: xml:test-results/
+
+jobs:
+  build:
+    # The CMake configure and build commands are platform agnostic anda should work equally well on Windows or Mac.
+    # You can convert this to a matrix build if you need cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-latest
+    env:
+      BOOST_PATH: ${{github.workspace}}/libraries
+      BOOST_ROOT: ${{github.workspace}}/libraries/boost/boost
+      BOOST_VERSION: 1.79.0
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install packages
+      run: |
+        wget https://github.com/red-panda-productions/ipc-lib/releases/download/v1.1.0/ipclib.deb
+        sudo apt install ./ipclib.deb
+        sudo apt-get install libplib-dev
+        sudo apt-get install libexpat1-dev
+
+    - name: Install GCovr
+      run: pip install gcovr
+
+    - name: Cache boost
+      uses: actions/cache@v2
+      id: cache-boost
+      with:
+        # Set the default path as the path to cache
+        path: ${{env.BOOST_ROOT}}
+        # Use the version as the key to only cache the correct version
+        key: boost-${{env.BOOST_VERSION}}
```

Comment:
Should this also contain the platform version? Now if the platform version is updated but not the boost version it will use the cache from the old platform